packages/node/src/integrations/http.ts

```diff
@@ -109,20 +105,29 @@ type WrappedRequestMethodFactory = (original: OriginalRequestMethod) => WrappedR
 function _createWrappedRequestMethodFactory(
   breadcrumbsEnabled: boolean,
   tracingEnabled: boolean,
-  tracePropagationTargets: TracePropagationTargets | undefined,
+  options: NodeClientOptions | undefined,
 ): WrappedRequestMethodFactory {
   // We're caching results so we dont have to recompute regexp everytime we create a request.
   const urlMap: Record<string, boolean> = {};
+
+  const shouldCreateSpan = (url: string): boolean => {
+    if (options?.shouldCreateSpanForRequest === undefined) {
+      return true;
+    }
+
+    return options.shouldCreateSpanForRequest(url);
+  };
```

Comment:
I think we should cache the results of `shouldCreateSpanForRequest`. On the browser side, [I can't decide if it's worth it](https://github.com/getsentry/sentry-javascript/pull/6041#pullrequestreview-1157439105), but in node where bundle size isn't a concern but being able to most efficiently handle lots of incoming requests quickly is, I think we definitely should (the same way we're doing with `tracePropagationTargets` just below).

Reply:
Since `lru_map` is already a dependency for the `ContextLines` integration, should I use that rather than an alternative that could potentially grow indefinitely?

Reply:
I just used a map for now!

Reply:
Sorry, missed your question. You raise two good points. I think for now a map is simple and hasn't bitten us on the header front, so is unlikely to bite us here, either. If it does, I'm sure someone will let us know! And good point about the docs.